pygmt/datasets/samples.py

```diff
@@ -123,3 +123,38 @@ def load_fractures_compilation():
     fname = which("@fractures_06.txt", download="c")
     data = pd.read_csv(fname, header=None, sep=r"\s+", names=["azimuth", "length"])
     return data[["length", "azimuth"]]
+
+
+def load_hotspots():
+    """
+    Load a table with the locations, names, and suggested icon sizes of
+    hotspots.
+
+    This is the ``@hotspots.txt`` dataset used in the GMT tutorials.
+
+    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
+    first time you invoke this function. Afterwards, it will load the data from
+    the cache. So you'll need an internet connection the first time around.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        The data table. Use ``print(data.describe())`` to see the available
+        columns.
+    """
+    fname = which("@hotspots.txt", download="c")
+    with open(fname) as hotspot_text:
+        hotspot_text.readline()
+        hotspot_text.readline()
+        hotspot_text.readline()
+        hotspots = []
+        for line in hotspot_text:
+            line_split = line.strip().split("\t")
+            # Add coordinates and icon_size of hotspot
+            hotspot = [float(item.strip()) for item in line_split[0].split()]
+            hotspot.append(line_split[1].title())  # Add name of hotspot
+            hotspots.append(hotspot)
+    data = pd.DataFrame(
+        hotspots, columns=["longitude", "latitude", "icon_size", "name"]
+    )
```

Comment:
```suggestion

    # Read first 3 columns (space-separated): longitude, latitude and icon_size
    lon_lat_size = pd.read_table(
        fname,
        sep="\s+",
        comment="#",
        usecols=[0, 1, 2],
        names=["longitude", "latitude", "icon_size"],
    )
    # Read last column (tab-separated): placename
    placename = pd.read_table(fname, comment="#", usecols=[1], names=["placename"])
    # Concat first 3 columns with last column to get a 4-column dataframe
    data = pd.concat(objs=[lon_lat_size, placename], axis="columns")

```

Reply:
This is way neater than my version! Nice work!